### PR TITLE
[cli] Add appIdSuffix arg to run android command and better handling of variant flag

### LINKED ISSUE #23266: Bare workflow android app with product flavors with different application id/package names fails to initialize activity
### Summary

I encountered an issue while working with Expo on Android, specifically when utilizing product flavors and changing the application ID. After modifying the application ID, the main activity of the app cannot be found, resulting in an error.

Expected Behavior:
The app should launch successfully with the modified application ID, and the main activity should be found and displayed.

Actual Behavior:
When attempting to launch the app with the modified application ID, the following error is encountered:
```
CommandError: No development build (com.bareflowproject) for this project is installed. Please make and install a development build on the device first.
```
If i go to the emulator i see the app there and i click in the app icon and it will be launched. 

### What platform(s) does this occur on?

Android

### SDK Version

48.0.15

### Environment

expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.0.1
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 18.16.0 - ~/.nvm/versions/node/v18.16.0/bin/node
      Yarn: 1.22.19 - ~/.nvm/versions/node/v14.17.0/bin/yarn
      npm: 9.5.1 - ~/.nvm/versions/node/v18.16.0/bin/npm
    Managers:
      CocoaPods: 1.12.1 - /Users/rafaelfecha/.rvm/gems/ruby-2.7.5/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.1, iOS 16.1, macOS 13.0, tvOS 16.1, watchOS 9.1
    IDEs:
      Android Studio: 2022.2 AI-222.4459.24.2221.10121639
      Xcode: 14.1/14B47b - /usr/bin/xcodebuild

### Minimal reproducible example
Github Repo: https://github.com/rafael-fecha/expo-bareflow-flavors-error

1. Install sample repo: npx react-native@latest init BareflowProject
2. Followed this guide to install expo and adapt the files: https://docs.expo.dev/bare/installing-expo-modules/#manual-installation
3. Add flavors in build gradle
```
flavorDimensions "env"
    productFlavors {
       production {
            dimension "env"
            applicationId = "com.bareflowproject"
        }
        dev {
            dimension "env"
            applicationId = "com.bareflowproject.dev"
        }
        staging {
            dimension "env"
            applicationId = "com.bareflowproject.staging"
        }
    }
```

### PR DESCRIPTION
**Pull request by prathameshmm02**
# Why

Fixes #23266. There were issues with run:android command for bare workflow, when dealing with flavors/variants in build.gradle. The path and apk file name generated by gradle did not match the ones resolved by expo cli.

Also, introduced appIdSuffix argument for the command which could be needed for apps with applicationIdSuffix in build.gradle.
# How

The bug has been fixed as per the docs of android regarding flavors: https://developer.android.com/build/build-variants#groovy
# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by manganapathy — COMMENTED** [review]
This looks Great @prathameshmm02 . Thanks for fixing this

File: packages/@expo/cli/src/run/android/resolveLaunchProps.ts [hunk 1]
```diff
@@ -25,16 +25,22 @@ async function getMainActivityAsync(projectRoot: string): Promise<string> {
...
+  options: { appIdSuffix?: string }
+): Promise<LaunchProps> {
   // Often this is ".MainActivity"
   const mainActivity = await getMainActivityAsync(projectRoot);
 
   const packageName = await new AndroidAppIdResolver(projectRoot).getAppIdFromNativeAsync();
+  const packageNameWithSuffix = options.appIdSuffix
+    ? `${packageName}.${options.appIdSuffix}`
+    : packageName;
   const launchActivity = `${packageName}/${mainActivity}`;
```
Comment (kuczi55) [inline packages/@expo/cli/src/run/android/resolveLaunchProps.ts:39]: Shouldn't `launchActivity` also be updated? I think in most cases it'll be `${packageNameWithSuffix}/${packageName}${mainActivity}`, e.g. cli from react-native-community prepares `activityToLaunch` in [such way](https://github.com/react-native-community/cli/blob/a414d98683f059162e9e106faab39f40ccf5228c/packages/cli-platform-android/src/commands/runAndroid/tryLaunchAppOnDevice.ts#L27) and later uses [${applicationIdWithSuffix}\/${activityToLaunch}](https://github.com/react-native-community/cli/blob/a414d98683f059162e9e106faab39f40ccf5228c/packages/cli-platform-android/src/commands/runAndroid/tryLaunchAppOnDevice.ts#L42) in `adbArgs`.

Moreover, `launchActivity` shouldn't be passed to [openCustomRuntimeAsync](https://github.com/expo/expo/blob/07d561b42396fd95452417899e775b993a384181/packages/%40expo/cli/src/run/android/runAndroidAsync.ts#L68)?
```ts
await manager.getDefaultDevServer().openCustomRuntimeAsync(
  'emulator',
  {
    applicationId: props.packageName,
    launchActivity: props.launchActivity
  },
  { device: props.device.device }
);
```

**Comment by manganapathy** [comment]
@EvanBacon @byCedric Kindly review this PR. This is a major blocker for us to enable expo run:android with product flavors

File: packages/@expo/cli/src/run/android/resolveLaunchProps.ts (hunk 1 above)
Comment (prathameshmm02) [reply packages/@expo/cli/src/run/android/resolveLaunchProps.ts]: Yes, you are right, I missed the case where app is opened from appId. I have borrowed the code from react-native about `launchActivity` and added `launchActivity` to `openCustomRuntimeAsync`.

**Pull request closed without merging**